fury/animation/timeline.py

```diff
@@ -0,0 +1,1499 @@
+
+    def remove_from_scene_at(self, timestamp):
+        """Set timestamp for removing Timeline to scene event.
+
+        Parameters
+        ----------
+        timestamp: float
+            Timestamp of the event.
+        """
+        if not self.is_interpolatable('in_scene'):
+            self.set_keyframe('in_scene', timestamp, False)
+            self.set_interpolator('in_scene', step_interpolator)
+        else:
+            self.set_keyframe('in_scene', timestamp, False)
+
+    def handle_scene_event(self, timestamp):
+        should_be_in_scene = self.is_inside_scene_at(timestamp)
+        if self._scene is not None:
+            if should_be_in_scene and not self._added_to_scene:
+                super(Timeline, self).add_to_scene(self._scene)
+                self._added_to_scene = True
+            elif not should_be_in_scene and self._added_to_scene:
+                super(Timeline, self).remove_from_scene(self._scene)
+                self._added_to_scene = False
+
+    def set_camera_keyframes(self, attrib, keyframes):
+        """Set multiple keyframes for a certain camera property
+
+        Parameters
+        ----------
+        attrib: str
+            The name of the property.
+        keyframes: dict
+            A dict object containing keyframes to be set.
+
+        Notes
+        ---------
+        Cubic Bézier curve control points are not supported yet in this setter.
+
+        Examples
+        ---------
+        >>> cam_pos = {1: np.array([1, 2, 3]), 3: np.array([5, 5, 5])}
+        >>> Timeline.set_camera_keyframes('position', cam_pos)
+        """
+        self.set_keyframes(attrib, keyframes, is_camera=True)
+
+    def set_interpolator(self, attrib, interpolator, is_camera=False,
+                         is_evaluator=False, **kwargs):
+        """Set keyframes interpolator for a certain property
+
+        Parameters
+        ----------
+        attrib: str
+            The name of the property.
+        interpolator: function
+            The generator function of the interpolator to be used to
+            interpolate/evaluate keyframes.
+        is_camera: bool, optional
+            Indicated whether dealing with a camera property or general
+            property.
+        is_evaluator: bool, optional
+            Specifies whether the `interpolator` is time-only based evaluation
+            function that does not depend on keyframes such as:
+            >>> def get_position(t):
+            >>>     return np.array([np.sin(t), np.cos(t) * 5, 5])
+
+        Other Parameters
+        ----------------
+        spline_degree: int, optional
+            The degree of the spline in case of setting a spline interpolator.
+
+        Notes
+        -----
+        If this evaluator is used to evaluate the value of actor's
+        properties such as position, scale, color, rotation, or opacity, it has
+        to return a value thathas the same shape as the evaluated property,
```

Comment:
If an evaluator is used to set the values of actor's properties...
thathas -> with